src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy

```diff
@@ -20,41 +34,74 @@ class BugsnagPlugin implements Plugin<Project> {
                 throw new IllegalStateException('Must apply \'com.android.application\' first!')
             }
 
-            // Create tasks for each variant
+            // Create tasks for each Build Variant
+            // https://sites.google.com/a/android.com/tools/tech-docs/new-build-system/user-guide#TOC-Build-Variants
             project.android.applicationVariants.all { ApplicationVariant variant ->
-                def variantName = variant.name.capitalize()
+                // The Android build system supports creating multiple APKs
+                // per Build Variant (Variant Outputs):
+                // https://sites.google.com/a/android.com/tools/tech-docs/new-build-system/user-guide/apk-splits
+                //
+                // Variant Outputs share most tasks so we only need to attach
+                // Bugsnag tasks to the first output.
                 def variantOutput = variant.outputs.first()
+                def variantName = variant.name.capitalize()
+
+                // The location of the "intermediate" AndroidManifest.xml
                 def manifestPath = variantOutput.processManifest.manifestOutputFile
 
                 // Create a Bugsnag task to add a "Bugsnag recommended proguard settings" file
-                BugsnagProguardConfigTask proguardConfigTask = project.tasks.create("processBugsnag${variant.name.capitalize()}Proguard", BugsnagProguardConfigTask)
+                BugsnagProguardConfigTask proguardConfigTask = project.tasks.create("processBugsnag${variantName}Proguard", BugsnagProguardConfigTask)
                 proguardConfigTask.group = GROUP_NAME
                 proguardConfigTask.applicationVariant = variant
-                proguardConfigTask.dependsOn variantOutput.processManifest
 
                 // Create a Bugsnag task to add a build UUID to AndroidManifest.xml
-                BugsnagManifestTask manifestTask = project.tasks.create("processBugsnag${variant.name.capitalize()}Manifest", BugsnagManifestTask)
+                // This task must be called after "process${variantName}Manifest", since it
+                // requires that an AndroidManifest.xml exists in `build/intermediates`.
+                BugsnagManifestTask manifestTask = project.tasks.create("processBugsnag${variantName}Manifest", BugsnagManifestTask)
                 manifestTask.group = GROUP_NAME
                 manifestTask.manifestPath = manifestPath
-                manifestTask.dependsOn variantOutput.processManifest
+                manifestTask.mustRunAfter variantOutput.processManifest
```

Comment:
`dependsOn` - go ahead and add this task as a dependency (adds the task to be run if it wasn't already)
`mustRunAfter` - if both of these tasks are told to execute, make sure X runs after Y (doesn't actually add a dependency)